Content.Server/PDA/Ringer/RingerSystem.cs

```diff
@@ -108,7 +108,30 @@ public void RandomizeRingtone(EntityUid uid, RingerComponent ringer, MapInitEven
 
         public void RandomizeUplinkCode(EntityUid uid, RingerUplinkComponent uplink, ComponentInit args)
         {
-            uplink.Code = GenerateRingtone();
+            var notes = new[]
+            {
+                Note.A,
+                Note.Asharp,
+                Note.B,
+                Note.C,
+                Note.Csharp,
+                Note.D,
+                Note.Dsharp,
+                Note.E,
+                Note.F,
+                Note.Fsharp,
+                Note.G,
+                Note.Gsharp,
+            };
+
+            var ringtone = new Note[RingtoneLength];
+
+            for (var i = 0; i < RingtoneLength; i++)
+            {
+                ringtone[i] = _random.Pick(notes);
+            }
+
+            uplink.Code = ringtone;
```

Comment:
this is just copypaste from GenerateRingtone, why not simply modify it?

Reply:
modification being either bool sharp or Note[] notes

Reply:
...as above

there's currently only 2 usecases, and that's standard ringtone generation and uplink code generation, which means you have a bool of sus and not sus already

could probably do that instead but need to include both of the lists anyway and IMO is probably better kept explicit (being *right next to each other*)

Reply:
Why isn't this just using a pentatonic anyway